include/extractor/guidance/turn_analysis.hpp

```diff
@@ -20,42 +23,61 @@ namespace extractor
 namespace guidance
 {
 
-struct TurnCandidate
+// What is exposed to the outside
+struct TurnOperation
+{
+    EdgeID eid;
+    double angle;
+    TurnInstruction instruction;
+};
+
+// For the turn analysis, we require a full list of all connected roads to determine the outcome.
+// Invalid turns can influence the perceived angles
+//
+// aaa(2)aa
+//          a - bbbbb
+// aaa(1)aa
+//
+// will not be perceived as a turn from (1) -> b, and as a U-turn from (1) -> (2).
+// In addition, they can influence whether a turn is obvious or not.
+struct ConnectedRoad : public TurnOperation
```

Comment:
Technically it contains a possible turn operation with a flag. We could set the Turn Operation as variable for that struct. Should not change much, but would elongate all accesses down the line.